crates/iceberg/src/scan.rs

```diff
@@ -151,42 +162,118 @@ impl TableScan {
                         ));
                     }
                     DataContentType::Data => {
-                        file_scan_tasks.push(Ok(FileScanTask {
-                            data_file: manifest_entry.clone(),
+                        plan_files.push(PlanFile {
+                            data_files: manifest_entry.data_file().clone(),
                             start: 0,
-                            length: manifest_entry.file_size_in_bytes(),
-                        }));
+                            len: manifest_entry.file_size_in_bytes(),
+                        });
                     }
                 }
             }
         }
 
-        Ok(iter(file_scan_tasks).boxed())
+        Ok(plan_files)
     }
 }
 
+/// PlanFile is a file to be scanned. It's the smallest unit of scan.
+#[derive(Debug)]
+pub struct PlanFile {
+    data_files: DataFile,
+    start: u64,
+    len: u64,
+}
+
 /// A task to scan part of file.
 #[derive(Debug)]
 #[allow(dead_code)]
 pub struct FileScanTask {
-    data_file: ManifestEntryRef,
-    start: u64,
-    length: u64,
+    plan_files: Vec<PlanFile>,
```

Comment:
The reason a `FileScanTask` is only a part of a data file is that in distributed computing environment, each `FileScanTask` will be distributed to different remote executors for execution.

Reply:
So does it mean that the FileScanTask can be serialized? The benefit of separating the plan file(metadata) is that we can make the plan file serializable without making the scan task serializable. It may make the scan task more flexible, e.g. contains some local state. But I'm not sure whether it's worth. For now, both way looks good to me.

Reply:
In fact, it's reasonable to make `FileScanTask` contains multi `PlanFile`, since in size based planning, each task may need to scan multi files, but I prefer to leave this to when we actually implement size based planning.

As with ser/deser, I think this is not a big problem, since we can always add new data structures to contains local states, and leave current `FileScanTask` as description only.